lib/Model.ts

```diff
@@ -86,6 +88,28 @@ export class Model<TDocument, TInstance> {
             this._Instance = instanceType.bind(undefined, this);
     }
 
+    /**
+     * Adds the validators to this Model's validators that are declared in
+     * all of the Plugin objects.
+     *
+     * Only Validators in the Plugins that have been registered with the Core
+     * will be added.
+     * 
+     * @private
+     * @memberof ModelHelpers
+     */
+    private loadPluginValidators(): void {
+        this.core.plugins.forEach((plugin) => {
```

Comment:
You could actually replace this entire function with `this._validators = [].concat(this.core.plugins.map(p => p.validate).filter(v => !!v)..., (instanceType.validators || []))` since `Array.prototype.concat` will flatten arrays it receives.

```javascript
[1,2].concat(3, [4,5], 6)
// => [1,2,3,4,5,6]
```

I also suspect it'd be nice to have the model's validators registered after the plugin validators - since that'll allow users to override the plugin validators on a per-model basis if they wish.

Reply:
Ah good point about doing the model validators after the plugin validators. I'll make that fix.
Also, your suggestion seems to me anyways a bit hard to read. But I didn't know about the concat able to do individual items and arrays. I'll clean mine up a bit but I'll still probably do it in steps. Like I'll do the `[].concat(this.core.plugins.map(p => p.validate).filter(v => !!v))` part and then in another statement concat `instanceType.validators || []`. Which removes the extra function but makes it readable and compact.